front/components/data_source_view/DataSourceViewSelector.tsx

```diff
@@ -161,84 +153,83 @@ export function DataSourceViewsSelector({
     managedDsv.length > 0 &&
     (useCase === "assistantBuilder" || useCase === "trackerBuilder");
 
-  const defaultSpace = useMemo(() => {
-    const firstKey = Object.keys(selectionConfigurations)[0] ?? null;
-    return firstKey
-      ? selectionConfigurations[firstKey]?.dataSourceView?.spaceId ?? ""
-      : "";
-  }, [selectionConfigurations]);
-
-  const filteredSpaces = useMemo(() => {
-    const spaceIds = [...new Set(dataSourceViews.map((dsv) => dsv.spaceId))];
-
-    return spaces.filter((s) => spaceIds.includes(s.sId));
-  }, [spaces, dataSourceViews]);
-
-  if (isSpacesLoading) {
-    return <Spinner />;
-  }
-
-  if (filteredSpaces.length > 1) {
-    return (
-      <SpaceSelector
-        spaces={filteredSpaces}
-        allowedSpaces={allowedSpaces}
-        defaultSpace={defaultSpace}
-        renderChildren={(space) => {
-          const dataSourceViewsForSpace = space
-            ? dataSourceViews.filter((dsv) => dsv.spaceId === space.sId)
-            : dataSourceViews;
-
-          if (dataSourceViewsForSpace.length === 0) {
-            return <>No data source in this space.</>;
-          }
-
-          return (
-            <DataSourceViewsSelector
+  return (
```

Comment:
diff here is hard to read, assume you did not change anything in the tree ?

Reply:
(keeping the "filteredSpaces" we had before)

Reply:
Nop didn't change anything here